src/test/ui/destructuring-assignment/struct_destructure.rs

```diff
@@ -0,0 +1,21 @@
+// run-pass
+
+#![feature(destructuring_assignment)]
+struct Struct<S, T> {
+    a: S,
+    b: T,
+}
+
+fn main() {
```

Comment:
We should make sure we can destructure unit structs for consistency (even though this is not useful).

Reply:
And single-variant enums, for that matter. We should check they behave properly with `#[non_exhaustive]`. This will be covered by the desugaring, but we need to make sure thisc can't regress.

Reply:
Working on it.